src/main/java/uk/co/bconline/ndelius/service/impl/DomainEventServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+package uk.co.bconline.ndelius.service.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.co.bconline.ndelius.model.entity.NDParameterEntity;
+import uk.co.bconline.ndelius.model.entity.ReferenceDataEntity;
+import uk.co.bconline.ndelius.model.notification.DomainEvent;
+import uk.co.bconline.ndelius.model.notification.HmppsDomainEvent;
+import uk.co.bconline.ndelius.model.notification.HmppsDomainEventType;
+import uk.co.bconline.ndelius.repository.db.DomainEventRepository;
+import uk.co.bconline.ndelius.repository.db.NDParameterRepository;
+import uk.co.bconline.ndelius.repository.db.ReferenceDataRepository;
+import uk.co.bconline.ndelius.service.DomainEventService;
+import uk.co.bconline.ndelius.transformer.DomainEventTransformer;
+
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Optional;
+
+@Slf4j
+@Service
+public class DomainEventServiceImpl implements DomainEventService
+{
+    private final ReferenceDataRepository referenceDataRepository;
+
+    private final DomainEventRepository domainEventRepository;
+    private final DomainEventTransformer domainEventTransformer;
+
+    private final NDParameterRepository ndParameterRepository;
+
+    private static final String ENABLE_DOMAIN_EVENTS_PARAM = "ENABLE_DOMAIN_EVENTS";
+    private static final String SELF_PUBLISH_DOMAIN_EVENTS_PARAM = "SELF_PUBLISH_DOMAIN_EVENTS";
```

Comment:
We can remove these feature flag checks to simplify this - they were only meant for Delius anyway.